tests/unittests/nominal/test_theils_u.py

```diff
@@ -0,0 +1,175 @@
+_preds[-1, -1] = float("nan")
+_target = torch.randint(high=NUM_CLASSES, size=(NUM_BATCHES, BATCH_SIZE), dtype=torch.float)
+_target[1, 0] = float("nan")
+_target[-1, 0] = float("nan")
+_input_with_nans = Input(preds=_preds, target=_target)
+
+_input_logits = Input(
+    preds=torch.rand(NUM_BATCHES, BATCH_SIZE, NUM_CLASSES), target=torch.rand(NUM_BATCHES, BATCH_SIZE, NUM_CLASSES)
+)
+
+
+@pytest.fixture
+def _matrix_input():
+    matrix = torch.cat(
+        [
+            torch.randint(high=NUM_CLASSES, size=(NUM_BATCHES * BATCH_SIZE, 1), dtype=torch.float),
+            torch.randint(high=NUM_CLASSES + 2, size=(NUM_BATCHES * BATCH_SIZE, 1), dtype=torch.float),
+            torch.randint(high=2, size=(NUM_BATCHES * BATCH_SIZE, 1), dtype=torch.float),
+        ],
+        dim=-1,
+    )
+    matrix[0, 0] = float("nan")
+    matrix[-1, -1] = float("nan")
+    return matrix
+
+
+def _dython_theils_u(preds, target, bias_correction, nan_strategy, nan_replace_value):
+    preds = preds.argmax(1) if preds.ndim == 2 else preds
+    target = target.argmax(1) if target.ndim == 2 else target
+
+    v = dython_theils_u(
+        preds.numpy(),
+        target.numpy(),
+        nan_strategy=nan_strategy,
+        nan_replace_value=nan_replace_value,
+    )
+    return torch.tensor(v)
+
+
+def _dython_theils_u_matrix(matrix, bias_correction, nan_strategy, nan_replace_value):
+    num_variables = matrix.shape[1]
+    theils_u_matrix_value = torch.ones(num_variables, num_variables)
+    for i, j in itertools.combinations(range(num_variables), 2):
+        x, y = matrix[:, i], matrix[:, j]
+        theils_u_matrix_value[i, j] = _dython_theils_u(x, y, bias_correction, nan_strategy, nan_replace_value)
+        theils_u_matrix_value[j, i] = _dython_theils_u(y, x, bias_correction, nan_strategy, nan_replace_value)
+    return theils_u_matrix_value
+
+
+@pytest.mark.skipif(
+    _compare_version("pandas", operator.lt, "1.3.2"), reason="`dython` package requires `pandas>=1.3.2`"
+)
+@pytest.mark.skipif(  # TODO: testing on CUDA fails with pandas 1.3.5, and newer is not available for python 3.7
+    torch.cuda.is_available(), reason="Tests fail on CUDA with the most up-to-date available pandas"
+)
+@pytest.mark.parametrize(
+    "preds, target",
+    [
+        (_input_default.preds, _input_default.target),
+        (_input_with_nans.preds, _input_with_nans.target),
+        (_input_logits.preds, _input_logits.target),
+    ],
+)
+@pytest.mark.parametrize("bias_correction", [False, True])
+@pytest.mark.parametrize("nan_strategy, nan_replace_value", [("replace", 0.0), ("drop", None)])
+class TestTheilsU(MetricTester):
+    atol = 1e-5
+
+    @pytest.mark.parametrize("ddp", [False, True])
+    @pytest.mark.parametrize("dist_sync_on_step", [False, True])
+    def test_theils_u(self, ddp, dist_sync_on_step, preds, target, bias_correction, nan_strategy, nan_replace_value):
+        metric_args = {
+            "bias_correction": bias_correction,
```

Comment:
Just drop `bias_correction` from metric args as the argument is not used for this metric.